File: Algo/LeadToOffer/18_MirrotOfBinaryTree.py
## Recursively exchange left and right subtree of a non-leaf node.
class Solution:
    # 返回镜像树的根节点
    def Mirror(self, root):
        if root is None:
            return

        temp = root.left
        root.left = root.right
        root.right = temp

        self.Mirror(root.left)
        self.Mirror(root.right)
        return

File: Algo/LeadToOffer/18_MirrotOfBinaryTree.cpp
// Recursively exchange left and right subtree of a non-leaf node.
class Solution {
public:
    void Mirror(TreeNode *pRoot) {
        if (!pRoot)
            return;

        swap(pRoot->left, pRoot->right);
        Mirror(pRoot->left);
        Mirror(pRoot->right);
        return;
    }
};
